resources/tests.mjs

```diff
@@ -1113,6 +1113,90 @@ Suites.push({
     ],
 });
 
+Suites.push({
+    name: "Responsive-Design",
+    url: "experimental/responsive-design/dist/index.html",
+    tags: ["responsive-design", "webcomponents", "experimental"],
+    async prepare(page) {
+        (await page.waitForElement("#content-iframe")).focus();
+    },
+    tests: [
+        new BenchmarkTestStep("LoadChatAndExpandRecipes", (page) => {
+            const iframeElement = page.querySelector("#content-iframe", [], true);
+            const resumePreviousChatBtn = iframeElement.querySelectorInShadowRoot("#resume-previous-chat-btn", ["cooking-app", "chat-window"]);
+            resumePreviousChatBtn.click();
+            page.layout();
+
+            // Navigate through the restaurants
+            const nextRestaurantBtn = iframeElement.querySelectorInShadowRoot("#next-restaurant-btn", ["cooking-app", "information-window"]);
+            const restaurantCards = iframeElement.querySelectorAllInShadowRoot("restaurant-card", ["cooking-app", "information-window"]);
+            const numOfRestaurantCards = restaurantCards.length - 1; // since 1 is already visible
+            for (let i = 0; i < numOfRestaurantCards; i++) {
+                nextRestaurantBtn.click();
+                page.layout();
+            }
+
+            // Expand recipes
+            const showMoreBtn = iframeElement.querySelectorAllInShadowRoot(".show-more-btn", ["cooking-app", "main-content", "recipe-grid"]);
+            for (const btn of showMoreBtn) {
+                btn.click();
+                page.layout();
+            }
+        }),
+        new BenchmarkTestStep("ReduceWidthIn5Steps", (page) => {
+            const iframeElement = page.querySelector("#content-iframe");
+            const widths = [768, 704, 640, 560, 480];
+
+            widths.forEach((width) => {
+                iframeElement.setWidth(`${width}px`);
+                page.layout();
```

Comment:
I think the layout doesn't work, possibly because the resize observer is async? 
I think you should try doing several steps instead of just one, so that the async part is properly captured.